staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/configmaplock.go

```diff
@@ -86,6 +86,9 @@ func (cml *ConfigMapLock) Update(ler LeaderElectionRecord) error {
 	if err != nil {
 		return err
 	}
+	if cml.cm.Annotations == nil {
+		cml.cm.Annotations = make(map[string]string)
```

Comment:
Based on the comment on line 80, the method should return an error if annotation doesn't exist.

OTOH, this won't happen because cml.cm is initialized at line 68, which populate the annotations.

Reply:
Looking at ConfigMapLock#Get :
```
	cml.cm, err = cml.Client.ConfigMaps(cml.ConfigMapMeta.Namespace).Get(cml.ConfigMapMeta.Name, metav1.GetOptions{})
	if err != nil {
		return nil, nil, err
	}
	if cml.cm.Annotations == nil {
```
It seems if the Update runs in parallel to Get() which is checking cml.cm.Annotations against nil, the scenario described in #87800 may happen.

Reply:
In Update, it seems we can error out if cml.cm.Annotations == nil.

Reply:
Per my comment in https://github.com/kubernetes/kubernetes/pull/84801#issuecomment-601905313, when `Update` fails, we store an empty `*v1.ConfigMap` in the "cm" field. It doesn't require a race or any concurrency. It's due to the mistaken assumption that `Update` would have returned a more complete struct value.